### ISSUE TITLE
impl GpioExt for GPIOF/GPIOG?

### ISSUE THREAD

**Issue by ohmyarch** [stm32-rs/stm32f1xx-hal#288]
Seems to be missing.

**Comment by ohmyarch** [comment]
Just found that there are no modules for gpiof/gpiog.

**Comment by TheZoq2** [comment]
Good catch (and sorry for the delay), I don't think there is a good reason for us not having that already. If you're interested, I could show you how to add the impls

**Comment by xieren58** [comment]
How to add the impls ? (for gpiof/gpiog...)

**Comment by xieren58** [comment]
@TheZoq2

**Comment by TheZoq2** [comment]
Sorry for the delay. I think all that is required is to add another one of these macros https://github.com/stm32-rs/stm32f1xx-hal/blob/master/src/gpio.rs#L1315 for the other peripherals. It probably also needs feature gating because IIRC those ports are only available on some of the chips

**Comment by xieren58** [comment]
@TheZoq2 Thanks!

**Comment by codytrey** [comment]
Seems that additional timers need to be implemented to fully use GPIOF/G. Since that only affects things like pwm, and adc, should this be it's own issue?

**Comment by ohmyarch** [comment]
Any progress?
